File: mixed_media/richest_customer_wealth/wealth.py
from random import randint

M, N = 3, 3
min_balance, max_balance = -10000, 100000

def create_accounts():
  accounts = [[randint(min_balance, max_balance) for _ in range(N)]
              for _ in range(M)]
  return accounts


def max_wealth(accounts):
  max_balance = -1
  for i in range(M):
    balance = 0
    for j in range(N):
      balance += accounts[i][j]
    if balance > max_balance:
      max_balance = balance
  return max_balance


def max_wealth_2(accounts):
  return sum(max(accounts, key=sum))


accounts = create_accounts()
print(accounts)
print(max_wealth(accounts))
print(max_wealth_2(accounts))

File: mixed_media/richest_customer_wealth/wealth.cpp
//#!/usr/bin/env g++ -std=c++17

#include <iostream>
#include <vector>
#include <algorithm>
#include <numeric>
#include <random>

#define M 4
#define N 5


int maxWealth(const std::vector<std::vector<int>>& accounts) {
    return std::transform_reduce(
        accounts.cbegin(),
        accounts.cend(),
        0,
        [](auto a, auto b) { return std::max(a, b); },
        [](const auto& row) { return std::reduce(row.cbegin(), row.cend()); }
    );
}

// Random MxN matrix
std::vector<std::vector<int>> generateRandomMatrix(int rows, int cols, int minVal, int maxVal) {
    std::random_device rd;
    std::mt19937 gen(rd());
    std::uniform_int_distribution<> dis(minVal, maxVal);

    std::vector<std::vector<int>> matrix(rows, std::vector<int>(cols));
    for (auto& row : matrix) {
        for (auto& elem : row) {
            elem = dis(gen);
        }
    }
    return matrix;
}


void printMatrix(const std::vector<std::vector<int>>& matrix) {
    for (const auto& row : matrix) {
        for (const auto& elem : row) {
            std::cout << elem << " ";
        }
        std::cout << '\n';
    }
}

int main() {
    int minVal = 1, maxVal = 100; // Random value range

    std::vector<std::vector<int>> accounts = generateRandomMatrix(M, N, minVal, maxVal);

    std::cout << "Generated " << M << "x" << N << " matrix (accounts):\n";
    printMatrix(accounts);

    int result = maxWealth(accounts);
    std::cout << "\nThe maximum wealth: " << result << '\n';

    return 0;
}
